packages/ui-react/src/containers/AccountModal/forms/ChooseOffer.tsx

```diff
@@ -1,119 +1,67 @@
-  }
-};
+import useQueryParam from '../../../hooks/useQueryParam';
 
 const ChooseOffer = () => {
-  const checkoutController = getModule(CheckoutController);
-  const accountController = getModule(AccountController);
-
   const navigate = useNavigate();
   const location = useLocation();
   const { t } = useTranslation('account');
-  const { setOffer } = useCheckoutStore(({ setOffer }) => ({ setOffer }), shallow);
-  const { isLoading, offerType, setOfferType, offers, offersDict, defaultOfferId } = useOffers();
-  const { subscription } = useAccountStore.getState();
-  const [offerSwitches, updateOffer] = useCheckoutStore((state) => [state.offerSwitches, state.updateOffer]);
-  const isOfferSwitch = useQueryParam('u') === 'upgrade-subscription';
-  const availableOffers = isOfferSwitch ? offerSwitches : offers;
-  const offerId = availableOffers[0]?.offerId || '';
-
-  const updateAccountModal = useEventCallback((modal: keyof AccountModals) => {
-    navigate(modalURLFromLocation(location, modal));
-  });
+  const isSwitch = useQueryParam('u') === 'upgrade-subscription'; // @todo: Use a store var instead?
+  const isPendingOffer = useAccountStore(({ pendingOffer }) => ({ isPendingOffer: !!pendingOffer }));
 
-  const chooseOfferSubmitHandler: UseFormOnSubmitHandler<ChooseOfferFormData> = useCallback(
-    async ({ offerId }, { setSubmitting, setErrors }) => {
-      const offer = offerId && offersDict[offerId];
+  const { isLoading, availableOffers, defaultOfferType, hasMultipleOfferTypes, chooseOffer, switchSubscription } = useOffers();
 
-      if (!offer) return setErrors({ form: t('choose_offer.offer_not_found') });
+  const checkoutUrl = modalURLFromLocation(location, 'checkout');
+  const upgradePendingUrl = modalURLFromLocation(location, 'upgrade-subscription-pending');
+  const upgradeSuccessUrl = modalURLFromLocation(location, 'upgrade-subscription-success');
+  const upgradeErrorUrl = modalURLFromLocation(location, 'upgrade-subscription-error');
 
-      if (isOfferSwitch) {
-        const targetOffer = offerSwitches.find((offer) => offer.offerId === offerId);
-        const targetOfferId = targetOffer?.offerId || '';
+  const { values, errors, submitting, setValue, handleSubmit, handleChange } = useForm<ChooseOfferFormData>({
+    initialValues: { selectedOfferType: defaultOfferType, selectedOfferId: undefined },
+    validationSchema: object().shape({
+      selectedOfferId: mixed<string>().required(t('choose_offer.field_required')),
+      selectedOfferType: mixed<OfferType>().required(t('choose_offer.field_required')),
+    }),
+    onSubmit: async ({ selectedOfferId }) => {
+      if (!selectedOfferId) return;
 
-        try {
-          await checkoutController.switchSubscription(targetOfferId, determineSwitchDirection(subscription));
-          // switching a subscription takes a bit longer to process
-          await accountController.reloadSubscriptions({ delay: 7500 });
+      chooseOffer.mutateAsync(selectedOfferId);
```

Comment:
No await? The following `switchSubscription` call is immediately called. Is this OK?